src/DIRAC/WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -435,6 +442,20 @@ def submitPilots(self):
             lifetime_secs = result["Value"]
             ce.setProxy(proxy, lifetime_secs)
 
+            # Get valid token id needed
+            if "Token" in ce.ceParameters.get("Tag", []):
```

Comment:
I am not sure I like this approach. Should not this based on the CEType? Or what if we do it always?

Reply:
This has nothing to do with the CEType. Different CETypes can use either certificates or tokens. But there can be a reasonable default. Now the default is no Token meaning using certificates

Reply:
Is this an information that we can get from BDII? CC @andresailer